client/app/dedicated/server/dedicated-server.controller.js

```diff
@@ -530,5 +534,22 @@ angular.module('App').controller('ServerCtrl', (
   }
 
   load();
+
+  // Upgrade to Advance
+  $scope.isEligibleForUpgrade = () => isEligibleForUpgrade();
+  $scope.trackUpgradeToAdvance = () => trackUpgradeToAdvance();
+  $scope.URLS = URLS;
+
+  function trackUpgradeToAdvance() {
+    atInternet.trackClick({
+      name: 'upgrade_rise_to_advance',
+      type: 'action',
+    });
+  }
+
+  function isEligibleForUpgrade() {
```

Comment:
Since you are defining your constants as an array you can safely use the [Array.prototype.includes()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) method.

```js
  function isEligibleForUpgrade() {
    return ELIGIBLE_FOR_UPGRADE.SUBSIDIARIES.includes($scope.user.ovhSubsidiary)
      && ELIGIBLE_FOR_UPGRADE.COMMERCIAL_RANGE.includes($scope.server.commercialRange);
  }
```

Reply:
Doesn't make sense to me. We are supposed to use `_.map` , `_.filter` and `_.reduce` but not `_.includes`  even though `map`, `filter` and `reduce` are `ES5` and ` Array.prototype.includes` is `ES7`, this is cuckoo

Reply:
That was what I was intented to do, but I remember a past comment (in another PR) saying to prioritize the use of lodash over native ES. So, what's the preferred way, since the two methods will works after all (with Babel) ? :)

Reply:
Once again, my opinion on the subject is that we already polyfill ES5+ functions so we don't need `Lodash` for this usage, we should only use it as a complement for missing functions in the Javascript standard library.

This eases the way you think about wether you should use Lodash or not:
- does this function exist in Javascript ? Yes, then use it. No, use Lodash

Because the current way to think is:
- does this function exists in JS ? Yes, then is it ES5+ ? Yes, then use Lodash, else use Javascript

I don't want to have to remember if every function is available for ES5+ or not :(

Reply:
I agree with Frederic there, we should use as much standard JS as possible, we're transpiling anyways.

Reply:
@FredericEspiau @euhmeuh I partially agree with you. If you are sure you have a defined value then ok but what if your param is `undefined` ? You have to check before calling your function etc and in some cases I think it could worsen readability